packages/block-editor/src/components/block-popover/index.js

```diff
@@ -77,30 +77,36 @@ function BlockPopover(
 		};
 	}, [ selectedElement ] );
 
-	const { isZoomOut, sectionRootClientId, getParentSectionBlock } = useSelect(
-		( select ) => {
-			const {
-				isZoomOut: isZoomOutSelector,
-				getSectionRootClientId,
-				getParentSectionBlock: getParentSectionBlockFn,
-			} = unlock( select( blockEditorStore ) );
-
-			const root = getSectionRootClientId();
-			return {
-				sectionRootClientId: root,
-				isZoomOut: isZoomOutSelector(),
-				getParentSectionBlock: getParentSectionBlockFn,
-			};
-		},
-		[]
-	);
+	const {
+		isZoomOut,
+		sectionRootClientId,
+		getParentSectionBlock,
+		getBlockOrder,
+	} = useSelect( ( select ) => {
+		const {
+			isZoomOut: isZoomOutSelector,
+			getSectionRootClientId,
+			getParentSectionBlock: getParentSectionBlockFn,
+			getBlockOrder: getBlockOrderFn,
+		} = unlock( select( blockEditorStore ) );
+
+		const root = getSectionRootClientId();
+		return {
+			sectionRootClientId: root,
+			isZoomOut: isZoomOutSelector(),
+			getParentSectionBlock: getParentSectionBlockFn,
+			getBlockOrder: getBlockOrderFn,
+		};
+	}, [] );
 
 	// These elements are used to position the zoom out view vertical toolbar
 	// correctly, relative to the selected section.
 	const rootSectionElement = useBlockElement( sectionRootClientId );
 	const parentSectionElement = useBlockElement(
 		getParentSectionBlock( clientId ) ?? clientId
 	);
+	const isSectionSelected =
+		getBlockOrder( sectionRootClientId ).includes( clientId );
```

Comment:
I'm wondering why we make these selections outside of a `useSelect`? I'm curious, is there a reason not to have a store subscription?

We now have an inconsistency between:

- `isZoomOut` - subscribed to store state changes
- other selectors which are not subscribed to store state changes

Reply:
If we move to the select then we must run that select on every clientId change otherwise it will have stale data. Which one is worse? This or adding client id to the select's deps?

Reply:
I guess this is one of those where the component will re-render when the clientId changes anyway so this will always be up to date.

I just wondered about the inconsistency in approach vs `isZoomOut`.

Let's leave "as is" for now as the performance is hard to gauge at this distance.